drift-transport-netty/src/main/java/com/facebook/drift/transport/netty/codec/ZlibTransformer.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (C) 2013 Facebook, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.drift.transport.netty.codec;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.ByteBufInputStream;
+import io.netty.buffer.ByteBufOutputStream;
+import io.netty.buffer.Unpooled;
+
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.util.zip.DeflaterOutputStream;
+import java.util.zip.InflaterInputStream;
+
+public class ZlibTransformer
+        implements ByteBufTransformer
+{
+    private static final int ZLIB_COMPRESSION_BUFFER_SIZE = 512;
+    public static final ZlibTransformer ZLIB_TRANSFORMER = new ZlibTransformer();
+
+    @Override
+    public ByteBuf transform(ByteBuf data)
+    {
+        ByteBuf output = Unpooled.buffer(ZLIB_COMPRESSION_BUFFER_SIZE);
+        try (DeflaterOutputStream outputStream = new DeflaterOutputStream(new ByteBufOutputStream(output))) {
+            data.readBytes(outputStream, data.readableBytes());
+        }
+        catch (IOException e) {
+            throw new UncheckedIOException(e);
+        }
+        finally {
+            data.release();
+        }
+        return output;
+    }
+
+    @Override
+    public ByteBuf untransform(ByteBuf data)
+    {
+        try (InflaterInputStream inputStream = new InflaterInputStream(new ByteBufInputStream(data))) {
+            ByteBuf output = Unpooled.buffer(ZLIB_COMPRESSION_BUFFER_SIZE);
+            while (inputStream.available() > 0) {
+                output.writeBytes(inputStream, ZLIB_COMPRESSION_BUFFER_SIZE);
+            }
+            return output;
+        }
+        catch (IOException e) {
+            throw new UncheckedIOException(e);
+        }
+        finally {
+            data.release();
```

Comment:
I would leave a comment in the interface declaration that this method releases the input buffer. That may not be obvious.

Reply:
Also add a unit test that verifies that all available transformers do so.